aiida/orm/nodes/data/structure.py

```diff
@@ -1861,6 +1844,8 @@ def _get_object_pymatgen_structure(self, **kwargs):
 
         if self.pbc != (True, True, True):
             raise ValueError('Periodic boundary conditions must apply in all three dimensions of real space')
+        if self.cell == _DEFAULT_CELL:
```

Comment:
With my suggested requirement that the cell can be unset ONLY if `pbc = [False, False, False]`, this is not needed as the line above catches it already

Reply:
Ah, note here and in a few more comments above: actually self.pbc is always set to a tuple, not a list, so indeed we need to check against `(False, False, False)` and not `[False, False, False]`.